### PR TITLE
Improved functionality for rowData manipulation

### LINKED ISSUE #6: Move functions from SCP.replication to QFeatures
`QFeatures` could benefit from some functions currently exported by `SCP.replication`:

* ~countUniqueFeatures~
* replaceRowDataCols
* longRowData

### PR DESCRIPTION
**Pull request by cvanderaa**
This PR adds two new functions (taken from `SCP.replication`): 
- `getRowData` extracts the `rowData` from one or more assays in a `QFeatures` object. The output is a single `DFrame` table that contains the `rowData` for all selected assays.
- `setRowData` function modifies columns in the `rowData` from one or more assays in `QFeatures` object. This function can also be used to remove columns in the `rowData`.

I have updated the documentation and added some unit tests. I also added a new section (*Accessors*) in the `QFeatures.Rmd` section that explicits the usage of the accessor functions. I would like to have a review of this new section.

The points that I think need attention or that remain to be done: 

- [x] Do we agree on the function names?
- [x] Review the new section in the vignette
- [x] Update the `NEWS.md` file and bump version

This PR is supposed to close [this issue](https://github.com/UCLouvain-CBIO/SCP.replication/issues/6) in `SCP.replication`

**Edit**

After discussion, we decide to call `getRowData` -> `rbindRowData` and decided that `setRowData` is better implement as a replace method `rowData<-`.

### CODE REVIEW

**Comment by cvanderaa** [comment]
The PR faces some errors that I don't have locally, probably because I still need to update R and Bioconductor on my computer. I'll fix that asap!

**Comment by lgatto** [comment]
See discussion notes.

File: vignettes/QFeatures.Rmd [hunk 1]
```diff
@@ -190,6 +188,98 @@ hl$tag <- c("126", "127N", "127C", "128N", "128C", "129N", "129C",
...
 ```
 
+# Manipulating feature metadata 
+
+The `QFeatures` package provides some utility functions that streamline
+the accession and manipulation of the feature metadata. 
+
+The feature metadata, more generally referred to as `rowData` in the
+Bioconductor ecosystem, is specific to each assay in a `QFeatures` 
+object. Therefore there are as many `rowData` tables as numbers of
```
Comment (lgatto) [inline vignettes/QFeatures.Rmd]: Therefore there are as many `rowData` tables as there are assays.

File: vignettes/QFeatures.Rmd [hunk 2]
```diff
@@ -190,6 +188,98 @@ hl$tag <- c("126", "127N", "127C", "128N", "128C", "129N", "129C",
...
+function. The function will automatically select the columns that are
+common to all selected assays. 
+
+```{r rbindRowData}
+rbindRowData(hl, i = c("peptides", "proteins"))
+```
+
+We can also replace and add columns in the `rowData`.  This requires
+to provide a `List` where the names of the `List` point to the assay 
+to replace and the elements of the `List` contain a `DataFrame` with
```
Comment (lgatto) [inline vignettes/QFeatures.Rmd]: the assays to be updated

Comment (lgatto) [reply vignettes/QFeatures.Rmd]: contain `DataFrame`s with

File: vignettes/QFeatures.Rmd [hunk 3]
```diff
@@ -190,6 +188,98 @@ hl$tag <- c("126", "127N", "127C", "128N", "128C", "129N", "129C",
...
+
+Then, we create the list and name the element `proteins` so that the 
+new data is added to the `rowData` of the `proteins` assay. To add 
+the list, we insert it back into the `rowData`. 
+
+```{r}
+rowData(hl) <- List(proteins = dF)
+```
+
+See here that the new variables are added to the `rowData` of the
```
Comment (lgatto) [inline vignettes/QFeatures.Rmd]: As shown below, the new `mean` and `sd` variables have been added to ...

File: vignettes/QFeatures.Rmd [hunk 4]
```diff
@@ -190,6 +188,98 @@ hl$tag <- c("126", "127N", "127C", "128N", "128C", "129N", "129C",
...
+rowData(hl)[["proteins"]]
+```
+
+Note that you can also replace an existing column in the `rowData` by
+naming the column name in the `DataFrame` after the column to replace.
+
+The `QFeatures` package also provides the `longFormat` function that 
+converts a `QFeatures` object into a long table. Long tables are very
+useful when using `ggplot2` for data visualization. For instance, 
+suppose we want to visualize the quantititation data (present in the
```
Comment (lgatto) [inline vignettes/QFeatures.Rmd]: quantitative data or quantitation data

**Review by lgatto — CHANGES_REQUESTED** [review]

**Comment by lgatto** [comment]
@cvanderaa  Could you also update the NEWS file

**Comment by lgatto** [comment]
I don't get what is happening in the action and why it fails. It looks like it's related to the methods package that somehow fails to find a method.

**Comment by cvanderaa** [comment]
I inserted your suggestions and adapted the NEWS file. I also bumped the version. If we can ignore the GHA issue, I'm ready for the merge.

**Pull request merged**